api-2.2/pom.xml

```diff
@@ -0,0 +1,102 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+
+	<parent>
+		<groupId>org.openmrs.module</groupId>
+		<artifactId>reporting</artifactId>
+		<version>1.19.0-SNAPSHOT</version>
+	</parent> 
+
+	<artifactId>reporting-api-2.2</artifactId>
+	<packaging>jar</packaging>
+	<name>reporting Module api 2.2</name>
+	<description>2.2 api project for reporting</description>
+	
+	<properties>
+		<openMRSVersion>2.2.0</openMRSVersion>
+	</properties>
+
+	<dependencies>
+		<dependency>
+			<groupId>${project.parent.groupId}</groupId>
+			<artifactId>${project.parent.artifactId}-api</artifactId>
+			<version>${project.parent.version}</version>
+			<scope>provided</scope>
+		</dependency>	
+		<dependency>
+			<groupId>${project.parent.groupId}</groupId>
+			<artifactId>${project.parent.artifactId}-api-tests</artifactId>
+			<version>${project.parent.version}</version>
+			<type>test-jar</type>
+			<scope>test</scope>
+		</dependency>  
+ 
+          <dependency>
+			<groupId>org.openmrs.test</groupId>
+			<artifactId>openmrs-test</artifactId>
+			<type>pom</type>
+			<scope>test</scope>
+			<version>${openMRSVersion}</version> 
+		  </dependency>  
+		<dependency>
+			<groupId>org.openmrs.api</groupId>
+			<artifactId>openmrs-api</artifactId>
+			<version>${openMRSVersion}</version>
+		</dependency>
+		
+		<dependency>
+			<groupId>org.openmrs.api</groupId>
+			<artifactId>openmrs-api</artifactId>
+			<type>test-jar</type>
+			<scope>test</scope>
+			<version>${openMRSVersion}</version>
+		</dependency>
+
+         <dependency>
+			<groupId>org.openmrs.module</groupId>
+			<artifactId>reportingcompatibility-api</artifactId>
+			<version>2.0.7-SNAPSHOT</version>
+			<scope>provided</scope>
+		</dependency>	 
+			<dependency>
+				<groupId>cglib</groupId>
+				<artifactId>cglib-nodep</artifactId>
+				<version>2.2</version>
+			</dependency> 
+			<dependency>
+    		    <groupId>org.codehaus.jackson</groupId>
+    			<artifactId>jackson-core-asl</artifactId>
+    			<version>1.9.13</version>
+			</dependency> 
+	</dependencies>
+
```

Comment:
The cglib and jackson dependencies here look like compile-time dependencies, so these libraries will be packaged in the module.  Can you explain the need for this?

Reply:
@mseaton wothout those dependencies , tests fail at run time ,with errors such as "Failed to load Application context" , "NoClassFoundDefError" , so the tests cant run when any of the above dependencies is missing

Reply:
@mozzy11 can we make them test-scoped depencencies?  Or do you think those should be packaged up with the reporting module omod?

Reply:
better we make them test scoped ,